### ISSUE TITLE
API: /api/v1/config/Interfaces

### ISSUE THREAD

**Issue by nqb** [inverse-inc/packetfence#4264]
Adding a new VLAN on a current interface failed:
```console
# /usr/local/pf/sbin/pfperl-api get -M POST /api/v1/config/interfaces -c '{"id":"eth0","netmask":"255.255.255.0","type":"vlan-registration","vlan":"153","ipaddress":"192.168.153.1","dhcpd_enabled":"1"}'
Use of uninitialized value $r in vec at /usr/share/perl5/vendor_perl/Mojolicious/Command/get.pm line 19.
Use of uninitialized value $r in scalar assignment at /usr/share/perl5/vendor_perl/Mojolicious/Command/get.pm line 19.
Use of uninitialized value in concatenation (.) or string at /usr/local/pf/lib/pf/UnifiedApi/Controller/Config/Interfaces.pm line 178.
Device ".153" does not exist.
Device "TestDomain" does not exist.
Device "TestDomain" does not exist.
ERROR: trying to add VLAN #0 to IF -:153:-  error: No such device
```
If I replace `"id":"eth0"` by `"id":"eth0.153"` and add `"master":"eth0", "name":"eth0"`, interface is created with following errors:
```console
# /usr/local/pf/sbin/pfperl-api get -M POST /api/v1/config/interfaces -c '{"id":"eth0.154","netmask":"255.255.255.0","type":"vlan-isolation","vlan":"154","ipaddress":"192.168.154.1","dhcpd
_enabled":"1", "master":"eth0","name":"eth0"}'   
Use of uninitialized value $r in vec at /usr/share/perl5/vendor_perl/Mojolicious/Command/get.pm line 19.
Use of uninitialized value $r in scalar assignment at /usr/share/perl5/vendor_perl/Mojolicious/Command/get.pm line 19.
Device "eth0.154" does not exist.                                    
Device "TestDomain" does not exist.                              
Device "TestDomain" does not exist.           
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 48.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 50.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 58.                                                                                              Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 66.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 75.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 85.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 87.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 89.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 91.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 93.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 95.                                                                                              Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 97.
Use of uninitialized value $net in string eq at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 99.
Use of uninitialized value $net in string eq at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 99.
Use of uninitialized value $net in pattern match (m//) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 101.
Use of uninitialized value $net in concatenation (.) or string at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 114.
could not parse  at /usr/local/pf/lib/pfappserver/Model/Config/System.pm line 73.
Use of uninitialized value in numeric eq (==) at /usr/share/perl5/vendor_perl/Net/Netmask.pm line 409.
Device "TestDomain" does not exist.
Device "TestDomain" does not exist.
Use of uninitialized value in string ne at /usr/local/pf/lib/pf/ConfigStore/Network.pm line 150.
```

**Comment by extrafu** [comment]
@julsemaan Please have a look at that!

**Comment by julsemaan** [comment]
I only got one of the warnings while creating or updating using the following line (just corrected a new line issue with the one above):
```
/usr/local/pf/sbin/pfperl-api get -M POST /api/v1/config/interfaces -c '{"id":"eth0.154","netmask":"255.255.255.0","type":"vlan-isolation","vlan":"154","ipaddress":"192.168.154.1","dhcpd_enabled":"1", "master":"eth0","name":"eth0"}'
```

My commit above addresses that warning and the only messages left are the device does not exist

@nqb, if that is still happening, then re-open and add other details specific to your environment like the OS + exact build on which you're running

**Comment by nqb** [comment]
Hello there,

It's not possible to create a new VLAN through GUI (also see #4370), API calls need to be adjust. From what I see `name` option is mandatory:
```console
# current API call from GUI (without "name" option)
$ /usr/local/pf/sbin/pfperl-api get -M POST /api/v1/config/interfaces -c '{"id":"eth0","netmask":"255.255.255.0","type":"other","vlan":"201","ipaddress":"192.168.201.1","dhcpd_enabled":"1"}'|jq
Device ".201" does not exist.                         
ERROR: trying to add VLAN #0 to IF -:201:-  error: No such device  
{
  "message": "Error in creating interface VLAN .201",
  "status": 500
}
# with "name" option
$ /usr/local/pf/sbin/pfperl-api get -M POST /api/v1/config/interfaces -c '{"id":"eth0","netmask":"255.255.255.0","type":"other","vlan":"201","ipaddress":"192.168.201.1","dhcpd_enabled":"1", "name":"eth0"}'|jq
Device "eth0.201" does not exist.
Use of uninitialized value $code in numeric ge (>=) at /usr/local/pf/lib/pf/error.pm line 51.
{
  "message": "Interface eth0.201 updated",
  "status": 200
}
```
From my point of view, `id` and `vlan` option should be enough to create a new VLAN on a current interface. I mean: `name` option should not be mandatory.

Environment:
* Nightly builds under Debian
* Docker container with one interface

**Comment by julsemaan** [comment]
The ID shouldn't be necessary and shouldn't be eth0, it should be eth0.201 which we could construct both the name and the VLAN with

I'd say to keep things fast, I'd say we adjust the frontend to send the expected attributes since the API call works fine but expects specific attributes

**Comment by julsemaan** [comment]
Pushing to 9.1 since this is polishing as the existing calls work

**Comment by fdurand** [comment]
We can't create a new vlan interface on the admin gui, i think it's important to fix that for 9.0.
